backend/src/test/java/de/laura/project/controller/TriviaControllerTest.java

```diff
@@ -39,45 +45,104 @@ private String getUrl() {
 
         TriviaApiDataAggregation mockedTriviaDataAggregationList = new TriviaApiDataAggregation(1, mockedTriviaApiDataList);
 
-        when(restTemplate.getForEntity(url, TriviaApiDataAggregation.class))
+        when(restTemplate.getForEntity(ApiUrl, TriviaApiDataAggregation.class))
                 .thenReturn(new ResponseEntity<>(mockedTriviaDataAggregationList, HttpStatus.OK));
 
+        TriviaQuestionSetDTO triviaQuestionSetDTO = new TriviaQuestionSetDTO(10, 12, "medium");
+
+        //WHEN
+        ResponseEntity<Void> response = testRestTemplate.postForEntity(getUrl(), triviaQuestionSetDTO, Void.class);
+
+        //THEN
+        assertThat(response.getStatusCode(), is(HttpStatus.OK));
+    }
+
+
+    @Test
+    @DisplayName("Checks whether user's answer is correct")
+
+    public void checkAnswerIsCorrect() {
+
+        //GIVEN
+        List<String> answerForTriviaQuestionSet = List.of("correct answer", "answerOne", "answerTwo", "answerThree");
+        List<String> anotherAnswerForTriviaQuestionSet = List.of("another correct answer", "answerOne", "answerTwo", "answerThree");
+
+        tempTriviaQuestionDB.getTriviaQuestionSetList().add(new TriviaQuestionSet(1, "This is an awesome question", "correct answer", answerForTriviaQuestionSet));
+        tempTriviaQuestionDB.getTriviaQuestionSetList().add(new TriviaQuestionSet(2, "This is another awesome question", "another correct answer", anotherAnswerForTriviaQuestionSet));
+
+        TriviaAnswerDTO triviaAnswerDTO = new TriviaAnswerDTO(1, "correct answer");
+
+        //WHEN
+        ResponseEntity<Boolean> response = testRestTemplate.postForEntity(getUrl() + "/answer", triviaAnswerDTO, Boolean.class);
+
+        //THEN
+        assertThat(response.getStatusCode(), is(HttpStatus.OK));
+        assertTrue(response.getBody());
+    }
+
+    @Test
+    @DisplayName("Checks whether user's incorrect answer returns false")
+
+    public void checkAnswerIsFalse() {
+
+        //GIVEN
+        List<String> answerForTriviaQuestionSet = List.of("correct answer", "answerOne", "answerTwo", "answerThree");
+        List<String> anotherAnswerForTriviaQuestionSet = List.of("another correct answer", "answerOne", "answerTwo", "answerThree");
+
+        tempTriviaQuestionDB.getTriviaQuestionSetList().add(new TriviaQuestionSet(1, "This is an awesome question", "correct answer", answerForTriviaQuestionSet));
+        tempTriviaQuestionDB.getTriviaQuestionSetList().add(new TriviaQuestionSet(2, "This is another awesome question", "another correct answer", anotherAnswerForTriviaQuestionSet));
+
+        TriviaAnswerDTO triviaAnswerDTO = new TriviaAnswerDTO(1, "answerOne");
+
+        //WHEN
+        ResponseEntity<Boolean> response = testRestTemplate.postForEntity(getUrl() + "/answer", triviaAnswerDTO, Boolean.class);
+
+        //THEN
+        assertThat(response.getStatusCode(), is(HttpStatus.OK));
+        assertFalse(response.getBody());
+    }
+
```

Comment:
you should consider including a test for posting an answer for a question that does not exist to ensure that your app returns an error code instead of crashing entirely